src/ImageList/index.js

```diff
@@ -104,19 +120,49 @@ export default class ImageList extends Component {
   }
 
   render() {
-    let { items } = this.props
+    let { items, searchBar } = this.props
 
     let layout = 'grid' //items[0] ? items[0].imageStyles.layout : 'grid'
 
+    const newItems = items.filter(
+      (itm) => itm.title.text.indexOf(this.state.currentQuery) >= 0
+    )
+
     if (layout === 'masonry') {
       return (
-        <View onLayout={this.handleLayout}>
-          {this.renderHeader()}
-          {this.renderMasonry()}
-        </View>
+        <>
+          <SearchBar
+            searchBar={this.props.searchBar}
+            onFilterElement={this.filterElement}
+          ></SearchBar>
+          {newItems.length == 0 ? (
+            <View style={([styles.input], { alignItems: 'center' })}>
+              {searchBar.notFoundText}
+            </View>
+          ) : (
+            <View onLayout={this.handleLayout}>
+              {this.renderHeader()}
+              {this.renderMasonry()}
```

Comment:
Because `renderMasonry` is what actually renders the list, you'll have to pass in `newItems` into the function. Then have `renderMasonry` use that instead of the list from props.